packages/twenty-front/src/modules/auth/components/VerifyEmailEffect.tsx

```diff
@@ -0,0 +1,64 @@
+import { useAuth } from '@/auth/hooks/useAuth';
+import { AppPath } from '@/types/AppPath';
+import { SnackBarVariant } from '@/ui/feedback/snack-bar-manager/components/SnackBar';
+import { useSnackBar } from '@/ui/feedback/snack-bar-manager/hooks/useSnackBar';
+
+import { useEffect, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { EmailVerificationSent } from '../sign-in-up/components/EmailVerificationSent';
+
+export const VerifyEmailEffect = () => {
+  const { verifyEmail } = useAuth();
+
+  const { enqueueSnackBar } = useSnackBar();
+
+  const [searchParams] = useSearchParams();
+  const [isError, setIsError] = useState(false);
+  const email = searchParams.get('email');
+  const emailVerificationToken = searchParams.get('emailVerificationToken');
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
```

Comment:
FYI we are trying to remove all useEffect in the codebase, but OK to keep this one for now  imo if it's too complex, since it's just a copy of the implementation we have in Verify. We should make sure to remove both in the future.

Reply:
Could you clarify the rationale behind removing useEffect across the codebase? Also what alternative pattern or strategy are we adopting to handle side effects and lifecycle events in its place?

Reply:
It becomes very messy if everyone uses useEffect everywhere, you can have re-render loops that are hard to debug. Here's an article that talks about alternatives: https://react.dev/learn/you-might-not-need-an-effect

Reply:
Got it, thanks!